mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/LinkBankAccountScreen.kt

```diff
@@ -0,0 +1,324 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.bank.presenter.BankUiState
+import org.mifos.mobilewallet.mifospay.bank.presenter.LinkBankAccountViewModel
+import org.mifos.mobilewallet.mifospay.designsystem.component.MfOverlayLoadingWheel
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosOutlinedTextField
+import org.mifos.mobilewallet.mifospay.designsystem.component.MifosTopBar
+import org.mifos.mobilewallet.mifospay.designsystem.theme.greyBackgroundColor
+import org.mifos.mobilewallet.mifospay.domain.model.Bank
+import org.mifos.mobilewallet.mifospay.theme.MifosTheme
+
+@Composable
+fun LinkBankAccountScreen(
+    viewModel: LinkBankAccountViewModel = hiltViewModel(),
+    context: Context = LocalContext.current,
+) {
+
+    var showSimBottomSheet by rememberSaveable { mutableStateOf(false) }
+    var bankSelected = ""
+
+
+    Scaffold(topBar = {
+        MifosTopBar(
+            topBarTitle = R.string.link_bank_account,
+            backPress = {
+                (context as Activity).onBackPressed()
+            }
+        )
+    }) {
+
+        LaunchedEffect(key1 = true) {
+            viewModel.getAllBanks(context)
+
+        }
+
+        val uiState by viewModel.bankState.collectAsState()
+
+
+        LinkBankAccountScreenContent(it, uiState, viewModel) {
+            bankSelected = it.name
+            showSimBottomSheet = true
+        }
+
+        if (showSimBottomSheet) {
+            ChooseSimDialogSheet { selectedSim ->
+                if (selectedSim != 0) {
+                    viewModel.fetchBankAccountDetails(bankSelected)
+                }
+                showSimBottomSheet = false
+            }
+        }
+    }
+}
+
+@Composable
+fun LinkBankAccountScreenContent(
+    padding: PaddingValues,
+    uiState: BankUiState,
+    viewModel: LinkBankAccountViewModel,
+    onSelectBank: (Bank) -> Unit
+) {
+
+
+    when (uiState) {
+        BankUiState.Loading -> {
+            MfOverlayLoadingWheel(
+                contentDesc = stringResource(R.string.loading),
+            )
+        }
+
+        is BankUiState.LoadedBankState -> {
+            LoadedContent(
+                popularBanks = uiState.popularBanks,
+                filteredBanks = uiState.allBanks,
+                viewModel,
+                onSelectBank
+            )
+        }
+    }
+
+
+}
+
+@Composable
+fun LoadedContent(
+    popularBanks: List<Bank>,
+    filteredBanks: List<Bank>,
+    viewModel: LinkBankAccountViewModel,
+    onSelectBank: (Bank) -> Unit
+
+
+) {
+
+    val search = remember {
+        mutableStateOf("")
+    }
```

Comment:
I assume this is for the search bar. In that case replicate the logic used in MerchantScreen and CardsScreen and make changes accordingly to your viewmodel